### ISSUE TITLE
radio buttons not working

### ISSUE THREAD

**Issue by danielhoover** [terminal42/contao-conditionalformfields#11]
Hey there Andreas and Yanick!

First of all congrats to that cool module that you've coded there.

I stumbled over an error that might be new (recent Version):

Following Setup:

Radio Button Menu (3 Options - radiobutton name tickettype) options: ticket1, ticket2, ticket3

Fieldset beginning (condition show when $tickettype == 'ticket1')

explanation for ticket1

fieldset end

fieldset beginning (condition show when $tickettype == 'ticket2')

explanation for ticket2

fieldset end

fieldset beginning (condition show when $tickettype == 'ticket3')

explanation for ticket 3

fieldset end

Now as I used the current Version (2.1.1) only the last radiobutton triggered to show the explanation for ticket 3. (no syntax error what so ever) I then installed an older Version (2.0.3 and 2.1.0 and tadaaa - everything worked as expected - with both versions).

Maybe something that was broken with the last hotfix?

Kind regards
Daniel

**Comment by aschempp** [comment]
Please tryto use `in_array` for radio buttons (see the [README](https://github.com/terminal42/contao-conditionalformfields/blob/hotfix/2.1.1/README.md))

**Comment by danielhoover** [comment]
Thank you Andreas! Unfortunately this didn't fix the issue:

here is what I did:

Replaced every:

Fieldset beginning (condition show when $tickettype == 'ticket1')

Accordingly with e.g. that:

Fieldset beginning (condition show when in_array('ticket1', $tickettype) )

Upgraded to latest Version - still same issue.

**Comment by aschempp** [comment]
Can you please try the hotfix/2.1.2 branch?

**Comment by TheRedTiger** [comment]
Hello, 
i´ve tryed the hotfix/2.1.2. - still same issue.

**Comment by aschempp** [comment]
@danielhoover what about you?

**Comment by danielhoover** [comment]
Hello Andreas,
no unfortunately that didn't solve the problem for me either.

If it is helpful here are the interesting parts of the generated html markup:

Following code are the 3 radiobuttons
```html
<div class="widget widget-radio mandatory">
        
      <fieldset id="ctrl_26" class="radio_container mandatory">

          <legend>
                  <span class="invisible">Pflichtfeld</span> Tickettyp<span class="mandatory">*</span>
              </legend>
    
    
          <span><input type="radio" name="tickettype" id="opt_26_0" class="radio" value="ticket1" required=""> <label id="lbl_26_0" for="opt_26_0">Ticket 1</label></span>
          <span><input type="radio" name="tickettype" id="opt_26_1" class="radio" value="ticket2" required=""> <label id="lbl_26_1" for="opt_26_1">Ticket 2</label></span>
          <span><input type="radio" name="tickettype" id="opt_26_2" class="radio" value="ticket3" required=""> <label id="lbl_26_2" for="opt_26_2">Ticket 3</label></span>
    
  </fieldset>

    </div>
```

Now the (shortened) Markup for the fieldsets that should be visible depending on which tickettype was selected:
```html
<fieldset class="cffs-27" style="display: none;">
  <div class="widget widget-explanation explanation">
    <h2>Explanation for Ticket 1…</h2>
 </div>
 […] and other fields […]
  </fieldset>

<fieldset class="cffs-32“ style="display: none;">
  <div class="widget widget-explanation explanation">
    <h2>Explanation for Ticket 2…</h2>
 </div>
 […] and other fields […]
  </fieldset>

<fieldset class="cffs-35“ style="display: none;">
  <div class="widget widget-explanation explanation">
    <h2>Explanation for Ticket 3…</h2>
 </div>
 […] and other fields […]
  </fieldset>
```

Finally the JavaScript generated at the End of the Page:
```html
<script>
(function($) {
    $('input[name="FORM_SUBMIT"][value="auto_form_1"]').parentsUntil('form').conditionalFormFields({
        'fields': ["tickettype"],
        'conditions': {"27":"return (in_array('ticket1', values.tickettype));","32":"return (in_array('ticket2', values.tickettype));","35":"return (in_array('ticket3', values.tickettype));"}
    });
})(jQuery);
</script>
```

In the Backend the radio Button Menu is defined as follows:

Fieldname: tickettype
Fieldlabel: Tickettyp
Fieldconfiguration -> required (Yes)

Options:
Value: ticket1 Label: Ticket 1
Value: ticket2 Label: Ticket 2
Value: ticket3 Label: Ticket 3


I inserted following condition for the 3 fieldsets in the Backend.

Fieldsetstart with ID 27:
```php
in_array('ticket1', $tickettype)
```

Fieldsetstart with ID 32:
```php
in_array('ticket2', $tickettype)
```

Fieldsetstart with ID 35:
```php
in_array('ticket3', $tickettype)
```

Hope this helps to hunt that nasty bug.

Kind regards
Daniel

**Comment by aschempp** [comment]
@qzminski can you take a look at this?

**Comment by aschempp** [comment]
But that solution would require to use `in_array` to check for radios, right?

**Comment by qzminski** [comment]
Can you guys try the latest fix from ```hotfix/2.1.2``` branch?

**Comment by basistom** [comment]
Works for me! Thank you!

**Comment by qzminski** [comment]
Fixed in ffe04d5.
